presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1170,6 +1172,14 @@ private void testPartitionPerScanLimit(Session session, HiveStorageFormat storag
                 "SHOW PARTITIONS FROM " + tableName + " WHERE part < 0",
                 "SELECT null WHERE false");
 
+        // using $partitions system table we can read beyond limit
```

Comment:
`// using $partitions system table we are not constrained by hive.max-partitions-per-scan`